hack/e2e/run-rp-and-e2e.sh

```diff
@@ -0,0 +1,122 @@
+            echo "local RP is NOT up $http_code, waiting"
+            sleep 2
+            ;;
+        esac
+    done
+}
+
+build_rp() {
+    echo "########## ⚙️ Building ARO RP Golang binary ##########"
+    go build ./cmd/aro
+}
+
+run_rp() {
+    echo "########## 🚀 Run ARO RP ##########"
+    ./aro rp &
+}
+
+deploy_e2e_db() {
+    echo "########## 📦 Creating new DB $DATABASE_NAME - $COSMOSDB_ACCOUNT in $RESOURCEGROUP ##########"
+
+    az group deployment create \
+      -g "$RESOURCEGROUP" \
+      -n "databases-development-test" \
+      --template-file deploy/databases-development.json \
+      --parameters \
+        "databaseAccountName=$COSMOSDB_ACCOUNT" \
+        "databaseName=$DATABASE_NAME" \
+        >/dev/null
+}
+
+deploy_e2e_deps() {
+    mkdir -p ~/.kube
+    RESOURCEGROUP=$CLUSTER_RESOURCEGROUP
+    CLUSTER=$CLUSTER_NAME
+    echo "🚀 Creating new RG: $RESOURCEGROUP and Vnet for cluster : $CLUSTER"
+
+    echo "########## Create ARO RG : $RESOURCEGROUP ##########"
+    az group create -g "$RESOURCEGROUP" -l $LOCATION >/dev/null
+
+    echo "########## Create ARO Vnet ##########"
+    az network vnet create \
+      -g "$RESOURCEGROUP" \
+      -n dev-vnet \
+      --address-prefixes 10.0.0.0/9 >/dev/null
+
+    echo "########## Create ARO Subnet ##########"
+    for subnet in "$CLUSTER-master" "$CLUSTER-worker"; do
+    az network vnet subnet create \
+        -g "$RESOURCEGROUP" \
+        --vnet-name dev-vnet \
+        -n "$subnet" \
+        --address-prefixes 10.$((RANDOM & 127)).$((RANDOM & 255)).0/24 \
+        --service-endpoints Microsoft.ContainerRegistry >/dev/null
+    done
+
+    echo "########## Update ARO Subnet ##########"
+    az network vnet subnet update \
+      -g "$RESOURCEGROUP" \
+      --vnet-name dev-vnet \
+      -n "$CLUSTER-master" \
+      --disable-private-link-service-network-policies true >/dev/null
+}
+
+register_sub() {
+    echo "########## 🔑 Registering subscription ##########"
+    curl -k -X PUT \
+      -H 'Content-Type: application/json' \
+      -d '{"state": "Registered", "properties": {"tenantId": "'"$AZURE_TENANT_ID"'"}}' \
+      "https://localhost:8443/subscriptions/$AZURE_SUBSCRIPTION_ID?api-version=2.0"
+}
+
+run_e2e() {
+    export RESOURCEGROUP=$CLUSTER_RESOURCEGROUP
+    export CLUSTER=$CLUSTER_NAME
+
+    echo "########## 🚀 Create ARO Cluster $CLUSTER ##########"
+    az aro create \
+      -g "$RESOURCEGROUP" \
+      -n "$CLUSTER" \
+      --vnet dev-vnet \
+      --master-subnet "$CLUSTER-master" \
+      --worker-subnet "$CLUSTER-worker"
+
+    echo "########## CLI : ARO List ##########"
+    az aro list -o table
+    echo "########## CLI : ARO list-creds ##########"
+    az aro list-credentials -g "$RESOURCEGROUP" -n "$CLUSTER"
+    echo "########## Run E2E ##########"
+    go run ./hack/kubeadminkubeconfig "/subscriptions/$AZURE_SUBSCRIPTION_ID/resourceGroups/$RESOURCEGROUP/providers/Microsoft.RedHatOpenShift/openShiftClusters/$CLUSTER" > ~/.kube/config
```

Comment:
my only minor quibble is to define and use KUBECONFIG rather than write over .kubeconfig as we use k8s/openshift for other things and over writing it is going to get annoying. anyways, can be a separate PR